### ISSUE TITLE
Register checkers in portal constructor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4595]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)| @moshez reported|
|-|-|
|Trac ID|trac#346|
|Type|enhancement|
|Created|2003-10-16 18:02:12Z|

<details><summary>Searchable metadata</summary>

```
trac-id__346 346
type__enhancement enhancement
reporter__moshez moshez
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066327332000000 1066327332000000
changetime__1468842378656596 1468842378656596
version__ 
owner__ 
cc__glyph cc__moshez
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Frequently you create a portal and you already know the checkers
you want to use. In those cases, it would be nice to register the
checkers right in the Portal initializor. This is what the patch
does.

Index: twisted/cred/portal.py
===================================================================
RCS file: /cvs/Twisted/twisted/cred/portal.py,v
retrieving revision 1.9
diff -u -r1.9 portal.py
--- twisted/cred/portal.py      26 Sep 2003 18:49:58 -0000      1.9
+++ twisted/cred/portal.py      16 Oct 2003 11:57:37 -0000
@@ -26,11 +26,13 @@
         pass

 class Portal:
-    def __init__(self, realm):
+    def __init__(self, realm, *checkers):
         """Create a Portal to a L{IRealm}.
         """
         self.realm = realm
         self.checkers = {}
+        for checker in checkers:
+            self.registerChecker(checker)

     def listCredentialsInterfaces(self):
         """Return list of credentials interfaces that can be used to login."""
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
/cvs/Twisted/twisted/cred/portal.py,v  &lt;--  portal.py
new revision: 1.10; previous revision: 1.9

Done.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset 229a9f522beaa19c6b288eaa0fc13294fa8606c0
```
#!CommitTicketReference repository="" revision="229a9f522beaa19c6b288eaa0fc13294fa8606c0"
Merge pull request #346 from twisted/8617-rodrigc-spread

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8617

Replace cStringIO with BytesIO
```
